hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/security/OzoneDelegationTokenSecretManager.java

```diff
@@ -71,17 +71,16 @@ public class OzoneDelegationTokenSecretManager
   private final S3SecretManager s3SecretManager;
   private Thread tokenRemoverThread;
   private final long tokenRemoverScanInterval;
-  private String omCertificateSerialId;
-  private String omServiceId;
+  private final String omServiceId;
   private final OzoneManager ozoneManager;
 
   /**
    * If the delegation token update thread holds this lock, it will not get
    * interrupted.
    */
-  private Object noInterruptsLock = new Object();
+  private final Object noInterruptsLock = new Object();
```

Comment:
Will it be same if we remove `noInterruptsLock` and use `this` [here](https://github.com/apache/ozone/blob/b2fe2a138ebd5101211c5aaaf14e695399307fb6/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/security/OzoneDelegationTokenSecretManager.java#L580)? I think it would be simpler to use `this`.

Reply:
We already have locked `this` at that point, since the method is `synchronized`.

https://github.com/apache/ozone/blob/4759560e6841157912c3f4e93756ead7ddd37620/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/security/OzoneDelegationTokenSecretManager.java#L557-L568

The deadlock happens when:

* `main` thread is holding lock on `OzoneDelegationTokenSecretManager` and waits for thread `ExpiredTokenRemover` to stop
* `ExpiredTokenRemover` waits to lock `OzoneDelegationTokenSecretManager`